components/form-input/src/FormInput.js

```diff
@@ -0,0 +1,202 @@
+/** @jsx jsx */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme } from '@westpac/core';
+
+import svgToTinyDataURI from 'mini-svg-data-uri';
+
+// ==============================
+// Utils
+// ==============================
+
+// ==============================
+// Component
+// ==============================
+
+export const FormInput = ({
+	size,
+	width,
+	invalid,
+	tag: Tag,
+	spacing,
+	inline,
+	children,
+	...props
+}) => {
+	const { colors, formInput, typography } = useTheme();
+
+	// Common styling
+	const styleCommon = {
+		display: inline ? 'inline-block' : 'block',
+		width: inline ? 'auto' : '100%',
+		appearance: 'none',
+		lineHeight: formInput.lineHeight,
+		fontWeight: formInput.fontWeight,
+		color: formInput.color,
+		backgroundColor: formInput.backgroundColor,
+		border: `${formInput.borderWidth}px solid`,
+		borderColor:
+			invalid || props.ariaInvalid
+				? formInput.appearance.invalid.borderColor
+				: formInput.appearance.default.borderColor,
+		borderRadius: formInput.borderRadius,
+		transition: 'border 0.2s ease',
+		verticalAlign: Tag === 'textarea' ? 'top' : inline ? 'middle' : null,
+
+		'&::placeholder': {
+			opacity: 1, // Override Firefox's unusual default opacity
+			...formInput.placeholder,
+		},
+
+		// Focus styling (for all, not just keyboard users)
+		':focus': {
+			...typography.link.focus,
+		},
+
+		// Focus styling (for all, not just keyboard users)
+		':focus': {
+			...typography.link.focus,
+		},
+
+		// Disabled and read-only inputs
+		':disabled, &[readonly]': {
+			cursor: 'not-allowed',
+			opacity: 1, // iOS fix for unreadable disabled content
+			...formInput.disabled,
+		},
+
+		// Disable number input spinners/steppers
+		'&::-webkit-outer-spin-button, &::-webkit-inner-spin-button': {
+			margin: 0,
+			appearance: 'none',
+		},
+	};
+
+	// Input size styling
+	const styleSize = {
```

Comment:
Move into common, dont need a separate object

Reply:
Sure was just keeping it consistent with some other components that separate styling relating to size, appearance etc props . But yeah not necessary